src/datadog_directive.cpp

```diff
@@ -1,380 +1,9 @@
-                  "the http block, before any "
-                  "proxy-related directives.",
-                  qualifier, &location.directive_name, &location.file_name,
-                  location.line);
-    return static_cast<char *>(NGX_CONF_ERROR);
-  }
-
-  const auto values = static_cast<ngx_str_t *>(cf->args->elts);
-  // values[0] is the command name, "datadog_propagation_styles".
-  // The other elements are the arguments: the names of the styles.
-  //
-  //     datadog_propagation_styles <style> [<styles> ...];
-  const auto args = values + 1;
-  const auto nargs = cf->args->nelts - 1;
-  auto &styles = main_conf->propagation_styles;
-  for (const ngx_str_t *arg = args; arg != args + nargs; ++arg) {
-    auto maybe_style = dd::parse_propagation_style(str(*arg));
-    if (!maybe_style) {
-      const auto location = command_source_location(command, cf);
-      ngx_log_error(NGX_LOG_ERR, cf->log, 0,
-                    "Invalid propagation style \"%V\". Acceptable values are "
-                    "\"Datadog\", \"B3\", "
-                    "and \"tracecontext\". Error occurred at \"%V\" in %V:%d",
-                    arg, &location.directive_name, &location.file_name,
-                    location.line);
-      return static_cast<char *>(NGX_CONF_ERROR);
-    }
-    if (std::find(styles.begin(), styles.end(), *maybe_style) != styles.end()) {
-      const auto location = command_source_location(command, cf);
-      ngx_log_error(NGX_LOG_ERR, cf->log, 0,
-                    "Duplicate propagation style \"%V\". Error occurred at "
-                    "\"%V\" in %V:%d",
-                    arg, &location.directive_name, &location.file_name,
-                    location.line);
-      return static_cast<char *>(NGX_CONF_ERROR);
-    }
-    styles.push_back(*maybe_style);
-  }
-
-  return lock_propagation_styles(command, cf);
-}
-
-char *set_datadog_agent_url(ngx_conf_t *cf, ngx_command_t *command,
-                            void *conf) noexcept {
-  assert(conf != nullptr);
-  auto &main_conf = *static_cast<datadog_main_conf_t *>(conf);
-
-  const auto values = static_cast<ngx_str_t *>(cf->args->elts);
-
-  // values[0] is the command name, while values[1] is the single argument.
-  const auto agent_url = to_string_view(values[1]);
-  if (agent_url.empty()) {
-    return static_cast<char *>(NGX_CONF_ERROR);
-  }
-
-  main_conf.agent_url = std::string(agent_url);
-  return NGX_OK;
-}
-
-char *warn_deprecated_command_1_2_0(ngx_conf_t *cf, ngx_command_t * /*command*/,
-                                    void * /*conf*/) noexcept {
-  const auto elements = static_cast<ngx_str_t *>(cf->args->elts);
-  assert(cf->args->nelts >= 1);
-
-  ngx_log_error(
-      NGX_LOG_WARN, cf->log, 0,
-      "Directive \"%V\" is deprecated and can be removed since v1.2.0.",
-      &elements[0]);
-
-  return NGX_OK;
-}
 
 char *warn_deprecated_command_datadog_tracing(ngx_conf_t *cf,
```

Comment:
Why are we letting that here and not in tracing/directives ?

Reply:
It could be used regardless of the product (rum, appsec or tracing). Think about it as a common function to handle deprecated directives.

Reply:
Good catch, that's what I did in otel PR.